src/lib/edge-functions/proxy.js

```diff
@@ -74,56 +78,62 @@ const initializeProxy = async ({
     projectDir,
   })
   const hasEdgeFunctions = userFunctionsPath !== undefined || internalFunctions.length !== 0
+  let hasServerError = false
 
   return async (req) => {
-    if (req.headers[headers.Passthrough] !== undefined || !hasEdgeFunctions) {
+    if (req.headers[headers.Passthrough] !== undefined || !hasEdgeFunctions || hasServerError) {
       return
     }
 
-    const [geoLocation, { registry }] = await Promise.all([
-      getGeoLocation({ mode: geolocationMode, offline, state }),
-      server,
-    ])
-
-    // Setting header with geolocation.
-    req.headers[headers.Geo] = JSON.stringify(geoLocation)
-
-    await registry.initialize()
-
-    const url = new URL(req.url, `http://${LOCAL_HOST}:${mainPort}`)
-    const { functionNames, orphanedDeclarations } = await registry.matchURLPath(url.pathname)
-
-    // If the request matches a config declaration for an Edge Function without
-    // a matching function file, we warn the user.
-    orphanedDeclarations.forEach((functionName) => {
-      log(
-        `${NETLIFYDEVWARN} Request to ${chalk.yellow(
-          url.pathname,
-        )} matches declaration for edge function ${chalk.yellow(
-          functionName,
-        )}, but there's no matching function file in ${chalk.yellow(
-          relative(cwd(), userFunctionsPath),
-        )}. Please visit ${chalk.blue('https://ntl.fyi/edge-create')} for more information.`,
-      )
-    })
-
-    if (functionNames.length === 0) {
-      return
-    }
-
-    req[headersSymbol] = {
-      [headers.Functions]: functionNames.join(','),
-      [headers.ForwardedHost]: `localhost:${mainPort}`,
-      [headers.Passthrough]: 'passthrough',
-      [headers.RequestID]: generateUUID(),
-      [headers.IP]: LOCAL_HOST,
+    try {
```

Comment:
I'm sorry if this wasn't clear in my previous comment, but I was trying to suggest that you only wrap the `Promise.all` call in the `try/catch`, not the entire block. Otherwise we're back to the problem of wrapping too much code and potentially swallowing errors.

Something like:

```ts
let ops

try {
  ops = await Promise.all([
    getGeoLocation({ mode: geolocationMode, offline, state }),
    server,
  ])
} catch (err) {
  error(err, { exit: false })

  return
}

const [geoLocation, { registry }] = ops
```

Reply:
Aaah I see, done.